homeassistant/components/homewizard_climate/__init__.py

```diff
@@ -0,0 +1,61 @@
+"""The Homewizard Climate integration."""
+from __future__ import annotations
+
+import logging
+
+from homewizard_climate_websocket.api.api import HomeWizardClimateApi
+from homewizard_climate_websocket.model.climate_device import HomeWizardClimateDevice
+from homewizard_climate_websocket.ws.hw_websocket import HomeWizardClimateWebSocket
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME, Platform
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS: list[Platform] = [Platform.CLIMATE]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Homewizard Climate from a config entry."""
+
+    api: HomeWizardClimateApi = HomeWizardClimateApi(
+        entry.data.get(CONF_USERNAME), entry.data.get(CONF_PASSWORD)
+    )
+    await hass.async_add_executor_job(api.login)
+
+    devices: list[HomeWizardClimateDevice] = await hass.async_add_executor_job(
+        api.get_devices
+    )
+    websockets = []
+    for device in devices:
+        websocket: HomeWizardClimateWebSocket = HomeWizardClimateWebSocket(
+            api,
+            device,
+            on_initialized=device_initialized,
+        )
+        websocket.connect_in_thread()
+        websockets.append(websocket)
+
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = {}
+    hass.data[DOMAIN][entry.entry_id]["websockets"] = websockets
+
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+
+    return True
+
+
+def device_initialized(device: HomeWizardClimateDevice):
+    """Todo."""
+    _LOGGER.info("Device %s initialized", device.identifier)
```

Comment:
Logging to info is only allowed for core, you should use `_LOGGER.debug()`
https://developers.home-assistant.io/docs/development_guidelines?_highlight=log#log-messages

Maybe it is not even needed to have this method, as successful initialisation is clearly shown because the device is 'available'

Reply:
Good to know, removed the callback altogether as I intended to do more in it initially but it doesn't make sense to keep it only for logging.